puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/tfmROR52.groovy

```diff
@@ -0,0 +1,67 @@
+def packages_to_build
+
+pipeline {
+    agent { label 'rpmbuild' }
+
+    options {
+        timestamps()
+        timeout(time: 2, unit: 'HOURS')
+        ansiColor('xterm')
+        buildDiscarder(logRotator(numToKeepStr: '15'))
+        disableConcurrentBuilds()
+    }
+
+    stages {
+        stage('Clone Packaging') {
+            steps {
+
+                deleteDir()
+
+                checkout([
+                    $class : 'GitSCM',
+                    branches : [[name: '*/tfm-ror52']],
+                    userRemoteConfigs: [
+                        [url: 'https://github.com/theforeman/rails-packaging']
+                    ]
+                ])
+
+            }
+        }
+        stage('Find packages') {
+            steps {
+
+                copyArtifacts(projectName: env.JOB_NAME, optional: true)
+
+                script {
+
+                    if (fileExists('commit')) {
+                        commit = readFile(file: 'commit').trim()
+
+                        if (commit) {
+                            packages_to_build = find_changed_packages("${commit}..HEAD")
+                        }
+                    } else {
+                        packages_to_build = 'all'
```

Comment:
Why build all in this case? Wouldn't we want this to be something more along the lines of none? Or is it more that Obal will safely ignore the packages that are already built, but will try to build anything that still needs building?

Reply:
Yea, it would try to fall back to build everything which has a check for existence in Koji.